packages/api/src/modules/predicate/services.ts

```diff
@@ -109,7 +114,37 @@ export class PredicateService implements IPredicateService {
         });
       }
 
-      return predicate;
+      const predicateProvider = predicate.provider;
+      const rawPredicateAddresss = Address.fromString(
+        predicate.predicateAddress,
+      ).toB256();
+
+      const deposits = await this.getPredicateHistory(
+        rawPredicateAddresss,
+        predicateProvider,
+      );
+
+      const getPredicateTransactions = await Transaction.createQueryBuilder('t')
+        .leftJoin('t.predicate', 'p')
+        .select(['t.id', 't.predicate', 't.hash', 'p.id', 't.createdAt'])
+        .where('p.id = :predicate', {
+          predicate: predicate.id,
+        })
+        .orderBy('t.createdAt', 'DESC')
+        .take(5)
+        .getMany();
+
+      const missingDeposits = deposits.filter(
+        deposit =>
+          !getPredicateTransactions.some(
```

Comment:
this is correct
must compare the transaction id returned from the graphql query with the hash column saved in the table

to compare, you will probably have to do:

`tx_graphql.id === 0x${tx_bakosafe.hash}`